src/main/java/org/team1540/robot2019/drivecontrol/commands/PurePursuitToVisionTarget.java

```diff
@@ -0,0 +1,132 @@
+package org.team1540.robot2019.drivecontrol.commands;
+
+import edu.wpi.first.wpilibj.command.Command;
+import org.apache.commons.math3.geometry.euclidean.threed.Vector3D;
+import org.apache.commons.math3.geometry.euclidean.twod.Vector2D;
+import org.apache.log4j.Logger;
+import org.team1540.robot2019.Hardware;
+import org.team1540.robot2019.Robot;
+import org.team1540.robot2019.Tuning;
+import org.team1540.robot2019.datastructures.threed.Transform3D;
+import org.team1540.robot2019.datastructures.twod.Twist2D;
+import org.team1540.robot2019.datastructures.utils.TrigUtils;
+import org.team1540.robot2019.odometry.TankDriveOdometryRunnable;
+import org.team1540.robot2019.utils.ControlUtils;
+import org.team1540.robot2019.utils.TankDriveTwist2DInput;
+import org.team1540.robot2019.vision.LimelightLocalization;
+import org.team1540.rooster.drive.pipeline.FeedForwardProcessor;
+import org.team1540.rooster.drive.pipeline.UnitScaler;
+import org.team1540.rooster.functional.Executable;
+
+// TODO: Use proper logging class
+// TODO: Make this a generic pure pursuit command
+public class PurePursuitToVisionTarget extends Command {
+
+    public static final Logger logger = Logger.getLogger(PurePursuitToVisionTarget.class);
+
+    private static final double ANGULAR_KP = 5;
+    private static final double LINEAR_KP = 4;
+    private static final double MAX_VEL_X = 0.8;
+    private static final double MIN_VEL_X = 0.2;
+    private static final double MAX_VEL_THETA = 2.0;
+    private static final double GOAL_DISTANCE_TOLERANCE = 0.05;
+    private static final Transform3D VISION_TARGET_OFFSET = new Transform3D(-0.65, -0.025, 0);
+
+    private final TankDriveOdometryRunnable driveOdometry;
+    private final LimelightLocalization limelightLocalization;
+    private Runnable onFail = null;
+
+    private Executable pipeline;
+    private TankDriveTwist2DInput twist2DInput;
+    private Transform3D goal;
+
+    public PurePursuitToVisionTarget(LimelightLocalization limelightLocalization, TankDriveOdometryRunnable driveOdometry) {
+        requires(Robot.drivetrain);
+        this.limelightLocalization = limelightLocalization;
+        this.driveOdometry = driveOdometry;
+        twist2DInput = new TankDriveTwist2DInput(Tuning.drivetrainRadiusMeters);
+        pipeline = twist2DInput
+            .then(new FeedForwardProcessor(0, 0, 0))
+            .then(new UnitScaler(Tuning.drivetrainTicksPerMeter, 10))
+            .then(Robot.drivetrain.getPipelineOutput());
+    }
+
+    public PurePursuitToVisionTarget(LimelightLocalization limelightLocalization, TankDriveOdometryRunnable driveOdometry, Runnable onFail) {
+        this(limelightLocalization, driveOdometry);
+        this.onFail = onFail;
+    }
+
+    @Override
+    protected void initialize() {
+        logger.info("Pure pursuit starting...");
```

Comment:
This stuff should probably be logged at `DEBUG`; `INFO` is sorta reserved for initialization stuff

Reply:
Done.